Libs/MRML/Widgets/qMRMLSliceControllerWidget.cxx

```diff
@@ -2306,6 +2375,65 @@ void qMRMLSliceControllerWidget::showReformatWidget(bool show)
     }
 }
 
+//---------------------------------------------------------------------------
+void qMRMLSliceControllerWidget::showSlabReconstructionWidget(bool show)
+{
+  Q_D(qMRMLSliceControllerWidget);
+
+  vtkSmartPointer<vtkCollection> nodes = d->saveNodesForUndo("vtkMRMLSliceNode");
+  if (!nodes.GetPointer())
+    {
+    return;
+    }
+
+  vtkMRMLSliceLogic* sliceLogic = nullptr;
+  vtkCollectionSimpleIterator it;
+
+  for (d->SliceLogics->InitTraversal(it);(sliceLogic = static_cast<vtkMRMLSliceLogic*>(
+                                               d->SliceLogics->GetNextItemAsObject(it)));)
+    {
+    if (sliceLogic == d->SliceLogic || this->isLinked())
+      {
+      vtkMRMLSliceNode* node = sliceLogic->GetSliceNode();
+
+      node->SetSlabReconstructionEnabled(show);
+
+      vtkImageReslice* resliceBG = sliceLogic->GetBackgroundLayer()->GetReslice();
+      vtkImageReslice* resliceFG = sliceLogic->GetForegroundLayer()->GetReslice();
+
+      double sliceSpacing;
+      if (node->GetSliceSpacingMode() == vtkMRMLSliceNode::AutomaticSliceSpacingMode)
+        {
+        sliceSpacing = *sliceLogic->GetLowestVolumeSliceSpacing();
+        }
+      else
+        {
+        sliceSpacing = node->GetPrescribedSliceSpacing()[2];
+        }
+
+      if (show)
+        {
+        int slabType = node->GetSlabReconstructionType();
+        resliceBG->SetSlabMode(slabType);
+        resliceFG->SetSlabMode(slabType);
+
+        int slabNumberOfSlices = int(node->GetSlabReconstructionThickness() / (sliceSpacing ? sliceSpacing : this->DefaultSlabReconstructionThickness));
+        resliceBG->SetSlabNumberOfSlices(slabNumberOfSlices);
+        resliceFG->SetSlabNumberOfSlices(slabNumberOfSlices);
+        }
+      else
+        {
+        resliceBG->SetSlabNumberOfSlices(this->DefaultSlabReconstructionThickness);
+        resliceFG->SetSlabNumberOfSlices(this->DefaultSlabReconstructionThickness);
+        }
+
+      double slabSliceSpacingFraction = sliceSpacing / node->GetSlabReconstructionOversamplingFactor();
+      resliceBG->SetSlabSliceSpacingFraction(slabSliceSpacingFraction);
+      resliceFG->SetSlabSliceSpacingFraction(slabSliceSpacingFraction);
```

Comment:
I am moving this into a new function called `vtkMRMLSliceLayerLogic::UpdateSlabReconstruction()` called when the associated node is modified.